zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/FileTxnSnapLog.java

```diff
@@ -611,14 +611,16 @@ public void rollLog() throws IOException {
      * @throws IOException
      */
     public void close() throws IOException {
```

Comment:
what's the purpose of the changes to this `close` method? I don't quite get the motivation behind this change.

Reply:
@hanm thanks for your review.

This little change is because during a graceful shutdown there is a race condition and multiple threads are going to call this method and we hit a NullPointerException that is usually harmless but it is very scary to see in the logs.

I hope you are fine with it

Reply:
i see - this also feels one of the those cases where a comment will help